### PR TITLE
Add a contribution workflow that spells out the gradle tasks needed

### PR DESCRIPTION
**Pull request by shartte**
Since I honestly have no idea where this information can be found otherwise, I'd like to describe step-by-step for someone unfamiliar with our tooling, how a PR is actually made (including the key info that we modify decompiled sources and run some magic tasks to get patches).

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -7,6 +7,24 @@ Contributing to NeoForge
 
 3) Follow the code style of the class you're working in (braces on newlines & spaces instead of tabs in Forge classes, inline brackets in patches, etc).
 
+## Workflow
+
+0. Have preliminary discussions on Discord (`#neoforge-github`)
+1. Fork the repository
+2. Check out your fork
+3. Make a branch
+4. Run `gradlew setup` from the project root
```
Comment (embeddedt) [inline docs/CONTRIBUTING.md]: `neoforge:setup` should be enough and avoids needing to decompile the base project too.

Comment (shartte) [reply docs/CONTRIBUTING.md]: Adjusted

Comment (Shadows-of-Fire) [reply docs/CONTRIBUTING.md]: Will `unpackSourcePatches` incur a decomp of the base project for diff purposes if it doesn't get decompiled during setup?  I'm not sure how it operates under the hood.

Comment (embeddedt) [reply docs/CONTRIBUTING.md]: Think I usually prefix that one with `neoforge:` too, come to think of it.

Comment (shartte) [reply docs/CONTRIBUTING.md]: I don't see `unpackSourcePatches` in base, so the prefix should not matter.

What is the base project actually used for?

Comment (Shadows-of-Fire) [reply docs/CONTRIBUTING.md]: @marchermans would know.  My understanding is that `unpackSourcePatches` generates the patches by doing a source diff between the `base` project and the `neoforge` project, which means that base must be setup as a prereq.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -7,6 +7,24 @@ Contributing to NeoForge
 
 3) Follow the code style of the class you're working in (braces on newlines & spaces instead of tabs in Forge classes, inline brackets in patches, etc).
 
+## Workflow
+
+0. Have preliminary discussions on Discord (`#neoforge-github`)
+1. Fork the repository
+2. Check out your fork
+3. Make a branch
+4. Run `gradlew :neoforge:setup` from the project root
```
Comment (Technici4n) [inline docs/CONTRIBUTING.md]: I would run `gradlew setup` which also sets up the `base` project.

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -7,6 +7,24 @@ Contributing to NeoForge
...
+5. Import project into your IDE (IntelliJ/Eclipse) or Reload Gradle Project
+6. Make your changes, modify the decompiled Minecraft source-code as needed
```
Comment (Technici4n) [inline docs/CONTRIBUTING.md]: I would mention that it is in `projects/neoforge/src`. And maybe add a note that the sources in the `base` project are there for reference and should not be modified.

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -7,6 +7,24 @@ Contributing to NeoForge
...
+7. Test your changes
+   - Run the game (Runs are available in the IDE)
+   - Run `gradlew :tests:runGameTestServer` or `Tests: GameTestServer` from IDE
+   - Run `gradlew :tests:runGameTestClient` or `Tests: GameTestClient` from IDE
+   - Check formatting with `gradlew spotlessCheck`
+8. Run `gradlew applyAllFormatting`
+9. Run `gradlew unpackSourcePatches`
```
Comment (Technici4n) [inline docs/CONTRIBUTING.md]: I would run `unpackSourcePatches` before `applyAllFormatting` because we have some custom spotless rules to check patches.

**Review by Technici4n — CHANGES_REQUESTED** [review]
We desperately needed this somewhere, thanks for doing it.

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (Shadows-of-Fire) [reply docs/CONTRIBUTING.md]: That's what the above discussion is on :p

File: docs/CONTRIBUTING.md (hunk 1 above)
Comment (marchermans) [reply docs/CONTRIBUTING.md]: No it is not needed.
The patches get generated from the unpatched jar in the neoforge subproject.

The base project is purely informative.

File: docs/CONTRIBUTING.md (hunk 4 above)
Comment (marchermans) [reply docs/CONTRIBUTING.md]: Yep these need to be flipped.

Comment (shartte) [reply docs/CONTRIBUTING.md]: Done

Comment (shartte) [reply docs/CONTRIBUTING.md]: I also moved checking of formatting with spotlessCheck to the end of the bullet point list.

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (shartte) [reply docs/CONTRIBUTING.md]: Done

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (shartte) [reply docs/CONTRIBUTING.md]: I switched it back to gradlew setup. "Optimizations" to the workflow can be done by peole more familiar with the project. For a first contributor, I suppose running gradlew setup and having both unmodified and modified sources is a better approach.

File: docs/CONTRIBUTING.md (hunk 1 above)
Comment (shartte) [reply docs/CONTRIBUTING.md]: I still changed it back to `gradlew setup` and added information that the sources in base are there for reference (and shouldn't be modified)

I think aimed at first contributors that at least explains what they are and that they should not be modified.

Comment (marchermans) [reply docs/CONTRIBUTING.md]: Yeah, any modifications on base don't matter, they don't flow into any patches and are overridden the next time.

**Review by Technici4n — APPROVED** [review]

**Review by marchermans — APPROVED** [review]

**Pull request merged**